lib/compat/wordpress-6.8/class-gutenberg-rest-post-types-controller-6-8.php

```diff
@@ -6,56 +6,36 @@
  */
 
 /**
- * Gutenberg_REST_Post_Types_Controller_6_8 class
+ * Register a REST field for the default rendering mode of a post type.
  *
- * Add Block Editor default rendering mode to the post type response
- * to allow enabling/disabling at the post type level.
+ * Note: Logic will become part of the `prepare_item_for_response` method when backporting to the core.
+ *
+ * @return void
  */
-class Gutenberg_REST_Post_Types_Controller_6_8 extends WP_REST_Post_Types_Controller {
-	/**
-	 * Add Block Editor default rendering mode setting to the response.
-	 *
-	 * @param  WP_Post_Type    $item    Post type object.
-	 * @param  WP_REST_Request $request Request object.
-	 * @return WP_REST_Response Response object.
-	 */
-	public function prepare_item_for_response( $item, $request ) {
-		$response = parent::prepare_item_for_response( $item, $request );
-		$context  = ! empty( $request['context'] ) ? $request['context'] : 'view';
-
-		// Property will only exist if the post type supports the block editor.
-		if ( 'edit' === $context && property_exists( $item, 'default_rendering_mode' ) ) {
-			/**
-			 * Filters the block editor rendering mode for a post type.
-			 *
-			 * @since 6.8.0
-			 * @param string       $default_rendering_mode Default rendering mode for the post type.
-			 * @param WP_Post_Type $post_type              Post type name.
-			 * @return string Default rendering mode for the post type.
-			 */
-			$rendering_mode = apply_filters( 'post_type_default_rendering_mode', $item->default_rendering_mode, $item );
+function gutenberg_editor_rendering_mode_field() {
+	register_rest_field( 'type', 'default_rendering_mode', array(
```

Comment:
It isn't well documented that `$object_type` in `register_rest_field` can refer to more than CPT, terms or comments.

Reply:
So what does `type` refer to here? 🤔  There is no "type" object type.

This is meant for post types, no?

```suggestion
	register_rest_field( 'post', 'default_rendering_mode', array(
```

Reply:
It refers to post-type objects. The rest fields reference for objects comes from item schema.

https://github.com/WordPress/wordpress-develop/blob/9104190b6d36c2959f652e0b65bf5bb7af42ffcc/src/wp-includes/rest-api/endpoints/class-wp-rest-controller.php#L527-L560